forest/intake_loader.py

```diff
@@ -0,0 +1,395 @@
+'''
+Module for loading data from the Pangeo CMIP6n intake catalogue (usng the intake-esm
+loader) created by NCAR.
+'''
+
+from datetime import datetime
+from collections import namedtuple
+import functools
+
+import bokeh
+import intake
+import iris
+import numpy
+
+from forest import geo, gridded_forecast
+
+# Location of the Pangeo-CMIP6 intake catalogue file.
+URL = 'https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmip6.json'
+
+@functools.lru_cache(maxsize=64)
+def _get_intake_vars(
+        experiment_id,
+        table_id,
+        grid_label,
+        institution_id,
+        member_id):
+    """
+    Helper function to get a list of variables for this particular combination
+    of parameters. Function is cahced to reduce remote queries.
+    """
+    collection = intake.open_esm_datastore(URL)
+    cat = collection.search(
+        experiment_id=experiment_id,
+        table_id=table_id,
+        grid_label=grid_label,
+        institution_id=institution_id,
+        member_id=member_id,
+    )
+    var_list = cat.unique('variable_id')['variable_id']['values']
+    return var_list
+
+
+@functools.lru_cache(maxsize=16)
+def _load_from_intake(
+        experiment_id,
+        table_id,
+        grid_label,
+        variable_id,
+        institution_id,
+        activity_id,
+        member_id):
+    """
+    Load data from the pangeo CMIP6 intake catalogue.The arguments relate to
+    the CMIP6 parameters of a dataset. The CMIP6 reference is the ESGF servers
+    which can be accessed here:
+    https://esgf-index1.ceda.ac.uk/search/cmip6-ceda/
+    Function is cahced to reduce remote queries.
+    """
+    collection = intake.open_esm_datastore(URL)
+    print('opening catalogue')
+    cat = collection.search(
+        experiment_id=experiment_id,
+        table_id=table_id,
+        grid_label=grid_label,
+        institution_id=institution_id,
+        member_id=member_id,
+        variable_id=variable_id)
+    print('downloading data')
+    dset_dict = cat.to_dataset_dict(
+        zarr_kwargs={'consolidated': True, 'decode_times': False},
+        cdf_kwargs={'chunks': {}, 'decode_times': False})
+    try:
+        parent_source_id = [l1 for l1 in dset_dict.values()][0].parent_source_id
+    except IndexError:
+        import pdb
+        pdb.set_trace()
```

Comment:
This section could do with some TLC ❤️

- We don't really want to call pdb 😉
- The list comprehension isn't doing anything - it's presumably a relic from an earlier state of the code?

Reply:
I've removed the debug statement at 4789a8d030da4bd85ea95e6a325a9b170e059c17 and updated this section of code.
It is a relic from when I though I need to get the parent source id to address the member of the data dictionary. I don't need to, I just need to get the first (and should be only) member of the dictionary and work on that, and I have updated the code to do that.